Tests/SwiftSyntaxMacroExpansionTest/PeerMacroTests.swift

```diff
@@ -193,4 +221,30 @@ final class PeerMacroTests: XCTestCase {
       ]
     )
   }
+
+  func testAddCompletionHandlerWhereThereIsNotAsync() {
+    assertMacroExpansion(
+      """
+      @addCompletionHandler
+      func f(a: Int, for b: String, _ value: Double) -> String { }
+      """,
+      expandedSource: """
+        func f(a: Int, for b: String, _ value: Double) -> String { }
+        """,
+      diagnostics: [
+        DiagnosticSpec(
+          message: "can only add a completion-handler variant to an 'async' function",
+          line: 2,
+          column: 1,
+          fixIts: [FixItSpec(message: "add 'async'")]
+        )
+      ],
+      macros: ["addCompletionHandler": AddCompletionHandler.self],
+      fixedSource: """
+        @addCompletionHandler
+        func f(a: Int, for b: String, _ value: Double) async-> String { }
```

Comment:
I assume we don't care about adding the proper trivia just in this test case? Though FWIW the completion handler macro seems like a fairly beefy macro to have in our tests in general.

Reply:
Should I make a smaller macro? 

The idea is to have a more real-show case in a test case, so developers can see an example and what the expected result is

Reply:
Examples seem like they would be better off living in the examples. The test should be the minimum of whatever we want to test. I don't really mind if you want to leave it for now (it was already there as far as I can tell), this was more of an off hand comment.

Having said that, if the idea is that this is an example, then that would presumably imply that we do care about the missing trivia 😅

Reply:
@bnbarham @ahoppen I've tried to reduce the test case. 
Should I try to reduce it more. It's properly not a valid real scenario, but it does the job in test I think

Reply:
Am I missing something or are you no longer adding a `fixedSource` parameter anymore? As it stands right now, I’m pretty sure the test will fail because we are no longer adding the completion handler.

What I would suggest is that we keep the completion handler test case for now because I’m not sure if it tests anything that’s not covered by the other peer macro test cases and you add the test that you wrote as a new test case that is intended to test the `fixedSource`.